### ISSUE TITLE
xml import: pickup bars and danish lyrics

### ISSUE THREAD

**Issue by Oldcamel86** [musescore/MuseScore#24602]
### Issue type

Import/export issue

### Description with steps to reproduce

1. Open the attached xml, which contains a pickup bar (1 beat) and i pickup bar again in measure 9 (1 beat). It also contains danish lyrics with letter ø and å.
2.  Compare the attached images. The picture from Sibelius is the same xml, imported into Sibelius. This is how the score should look.
I have marked the relevant places with red circles.

### Supporting files, videos and screenshots

[Pickup bars and danish lyrics.xml.zip](https://github.com/user-attachments/files/16934870/Pickup.bars.and.danish.lyrics.xml.zip)
<img width="631" alt="Import into Sibelius" src="https://github.com/user-attachments/assets/e64378ab-e06b-4507-bc14-f86cd688f509">
<img width="647" alt="Import Musescore 4 4 1" src="https://github.com/user-attachments/assets/4e13f706-4d79-4383-b15e-fca96eef66cd">

### What is the latest version of MuseScore Studio where this issue is present?

4.4.1

### Regression

I was unable to check

### Operating system

Mac OS 14.4.1, maybe others

### Additional context

Origin of xml is lesser known notation software Primus (www.columbussoft.de). Even though xml might not be a fixed standard, it would be nice, if Musescore could handle the pickup bars correct.

### Checklist

- [X] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [X] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [X] I have attached all requested files and information to this report
- [X] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by Jojo-Schmitz** [comment]
'''xml
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 1.1 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">
```
MusicXML 1.1, Pretty antediluvian...

**Comment by lvinken** [comment]
With respect to the Danish characters, that is a 4.4 regression, 4.3 imports those correctly. Note that the file is encoded in ISO-8859-1. Does the new muse::XmlStreamReader support that ?

**Comment by cbjeukendrup** [comment]
> ISO-8859-1. Does the new muse::XmlStreamReader support that ?

No, it doesn't, unfortunately. Only UTF 8 and two variants of UTF 16, at the moment.

**Comment by lvinken** [comment]
The incorrect pickup bar is caused by the empty measures in the second part. MuseScore's MusicXML importer assumes the length of empty measures to be equal to the length according to the current time signature. This works OK for single part scores, but fails for this one. Fixing this could be done in MusicXMLParserPa
ss1::determineMeasureLength(). Attached file [Pickup.bars.and.danish.lyrics.fixed.xml.zip](https://github.com/user-attachments/files/17070137/Pickup.bars.and.danish.lyrics.fixed.xml.zip) contains rests in the pickup bars in part two, it imports correctly.
